docs: add FAQ about updating libraries to static query migration guide (#31447)
angular/angular: aio/content/guide/static-query-migration.md

Yes, absolutely! 
Because we have not changed the default query behavior in version 8 (i.e. the compiler still chooses a timing if no flag is set), when your application runs with a library that has not updated to version 8, the library will run the same way it did in version 7.
This guarantees your app will work in version 8 even if libraries take longer to update their code.  

{@a update-library-to-use-static-flag}
###  Can I update my library to version 8 by adding the static flag to view queries, while still being compatible with Angular version 7 apps?

> Yes, the Core Team's recommendation for libraries is to update to version 8 and add the static flag. Angular version 7 apps will continue to work with libraries that have this flag.

Reviewer: Oh, I forgot to mention this earlier. Can you put static in backticks? So it's `static`.

Reviewer: Done.

> Yes, the Core Team's recommendation for libraries is to update to version 8 and add the `static` flag. Angular version 7 apps will continue to work with libraries that have this flag.

Reviewer 2: Nit: We're trying to get away from saying "core team"

```suggestion
Yes, the Angular team's recommendation for libraries is to update to version 8 and add the `static` flag. Angular version 7 apps will continue to work with libraries that have this flag.
```

Reviewer: Ah, that was me. Good to know!

> If you update your library to Angular version 8 and want to take advantage of the new version 8 APIs, or you want more recent dependencies, such as (Typescript or RxJS), your library will be incompatible with Angular version 7 apps. If your goal is to make your lib compatible with Angular versions 7 and 8, do not update your job at all—except for `peerDependencies` in `package.json`.

Reviewer 2: ```suggestion
However, if you update your library to Angular version 8 and want to take advantage of the new version 8 APIs, or you want more recent dependencies, such as (Typescript or RxJS), your library will become incompatible with Angular version 7 apps. If your goal is to make your lib compatible with Angular versions 7 and 8, you should not update your lib at all—except for `peerDependencies` in `package.json`.
```

> However, if you update your library to Angular version 8 and want to take advantage of the new version 8 APIs, or you want more recent dependencies, such as (Typescript or RxJS), your library will become incompatible with Angular version 7 apps. If your goal is to make your library compatible with Angular versions 7 and 8, you should not update your lib at all—except for `peerDependencies` in `package.json`.

Reviewer 2: ```suggestion
However, if you update your library to Angular version 8 and want to take advantage of the new version 8 APIs, or you want more recent dependencies, (such as Typescript or RxJS), your library will become incompatible with Angular version 7 apps. If your goal is to make your library compatible with Angular versions 7 and 8, you should not update your lib at all—except for `peerDependencies` in `package.json`.
```

> However, if you update your library to Angular version 8 and want to take advantage of the new version 8 APIs, or you want more recent dependencies, (such as Typescript or RxJS), your library will become incompatible with Angular version 7 apps. If your goal is to make your library compatible with Angular versions 7 and 8, you should not update your lib at all—except for `peerDependencies` in `package.json`.

Author: @kapunahelewong @kara just to double-check: in french language there is no sense to  write sentence then put comma then parenthesis then comm like here "...dependencies, (such as Typescript or RxJS), your library..." you confirm this is correct in english?

Reviewer: You are correct, @elvisbegovic. It would be one or the other, but not a comma along with the `()`. Maybe that got in there over the course of us editing it. Good catch though.

> For this reason, the most efficient plan is for libraries to adopt a 6 month major version schedule and bump their major after each Angular update. That way the libs are in the same release cadence as Angular. The automated migration via `ng update` will greatly assist with the transition and won't break version 7 libs.

Reviewer 2: ```suggestion
In general, the most efficient plan is for libraries is to adopt a 6 month major version schedule and bump the major version after each Angular update. That way, libraries stay in the same release cadence as Angular. 
```
